File: 백준/Silver/15654. N과 M （5）/N과 M （5）.cc
#include <bits/stdc++.h>

using namespace std;

int n,m;

int arr[10];

int arr2[10];

bool vis[200000];

void cur(int t){

    if(t == m){

        for(int i=0; i<m; i++){

            cout << arr2[i] << ' ';

        }

        cout << '\n';

        return;

    }

    

    for(int i=0; i<n; i++){

        if(!vis[arr[i]]){

            vis[arr[i]] = 1;

            arr2[t] = arr[i];

            cur(t+1);

            vis[arr[i]] = 0;

        }

    }

}

int main() {

    ios::sync_with_stdio(0);

    cin.tie(0);

    

    cin >> n >> m;

    for(int i=0; i<n; i++){

        cin >> arr[i];

    }

    sort(arr, arr+n);

    cur(0);

}

File: 백준/Silver/15654. N과 M （5）/N과 M （5）.py
import sys

n,m = map(int, input().split())
arr = list(map(int, input().split()))
arr.sort()

vis = [False] * 10
printArr = [''] * 10

def recur(k):
    if k == m:
        for i in range(k):
            print(printArr[i], end=' ')
        print()
    else:
        for i in range(n):
            if vis[i] == False:
                vis[i] = True
                printArr[k] = arr[i]
                recur(k+1)
                vis[i] = False


recur(0)